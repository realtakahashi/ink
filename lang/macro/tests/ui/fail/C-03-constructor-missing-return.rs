use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod noop {
    #[ink(storage)]
    struct Noop {}

    impl Noop {
        #[ink(constructor)]
        fn missing_return() {}

        #[ink(message)]
        fn noop(&self) {}
    }
}

fn main() {}
