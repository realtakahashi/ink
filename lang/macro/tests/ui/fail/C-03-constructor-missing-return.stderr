error: #[ink(constructor)] functions must have `Self` return type
 --> $DIR/C-03-constructor-missing-return.rs:3:1
  |
3 | #[ink::contract(version = "0.1.0")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
